### PR TITLE
Make video capture behave like photo capture re. storage options

### PR DESCRIPTION
**Pull request by akalin-keybase**
This makes video work on Android without storage permissions (the default). It also
prevents videos from showing up in Google Photos on some phones.

### CODE REVIEW

File: android/src/main/java/com/imagepicker/utils/MediaUtils.java [hunk 1]
```diff
@@ -66,6 +66,35 @@
         return result;
     }
 
+    public static @Nullable File createNewVideoFile(@NonNull final Context reactContext,
+                                                    @NonNull final ReadableMap options,
+                                                    @NonNull final boolean forceLocal)
+    {
+        final String filename = new StringBuilder("video-")
+                .append(UUID.randomUUID().toString())
+                .append(".mp4")
```
Comment (akalin-keybase) [inline android/src/main/java/com/imagepicker/utils/MediaUtils.java]: I'm assuming that most video files will be mp4, like `createNewFile` assumes most photo files will be jpg.

File: android/src/main/java/com/imagepicker/utils/MediaUtils.java [hunk 2]
```diff
@@ -66,6 +66,35 @@
         return result;
     }
 
+    public static @Nullable File createNewVideoFile(@NonNull final Context reactContext,
```
Comment (akalin-keybase) [inline android/src/main/java/com/imagepicker/utils/MediaUtils.java:47]: Ideally, createNewFile and createNewVideoFile would be combined, but this is a quick fix.

File: android/src/main/java/com/imagepicker/utils/RealPathUtil.java [hunk 3]
```diff
@@ -195,8 +195,16 @@ public static boolean isFileProviderUri(@NonNull final Context context,
 	public static @Nullable String getFileProviderPath(@NonNull final Context context,
 	                                                   @NonNull final Uri uri)
 	{
-		final File appDir = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
-		final File file = new File(appDir, uri.getLastPathSegment());
-		return file.exists() ? file.toString(): null;
+		final File photosDir = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
```
Comment (akalin-keybase) [inline android/src/main/java/com/imagepicker/utils/RealPathUtil.java]: Kinda gross that I had to do this.

**Comment by akalin-keybase** [comment]
r? @keybase/react-hackers 

Don't think this warrants a hot fix, since the current behavior is a black bar (and was present in the last release, too).

**Review by chrisnojima — APPROVED** [review]

**Comment by akalin-keybase** [comment]
I'll merge this and test after the release

**Comment by akalin-keybase** [comment]
actually nm, i think we kinda want this for the release

File: android/src/main/java/com/imagepicker/utils/MediaUtils.java (hunk 2 above)
Comment (akalin-keybase) [reply android/src/main/java/com/imagepicker/utils/MediaUtils.java:47]: Actually, just did this

**Comment by akalin-keybase** [comment]
Okay, made some fixes and cleaned up a bit. Merging.

**Pull request merged**